test/integration/base_integration_test.cc

```diff
@@ -694,4 +698,45 @@ AssertionResult BaseIntegrationTest::compareDeltaDiscoveryRequest(
   return AssertionSuccess();
 }
 
+void BaseIntegrationTest::checkForMissingTagExtractionRules() {
+  BufferingStreamDecoderPtr response = IntegrationUtil::makeSingleRequest(
+      test_server_->adminAddress(), "GET", "/config_dump", "", Http::CodecType::HTTP1);
+  ASSERT_TRUE(response->complete());
+  Json::ObjectSharedPtr json = Json::Factory::loadFromString(response->body());
+  std::vector<std::string> stat_prefixes;
+  Json::ObjectCallback find_stat_prefix = [&](const std::string& name,
+                                              const Json::Object& root) -> bool {
+    if (name == "stat_prefix") {
```

Comment:
where does this magic `"stat_prefix"` come from? Could this be a string-constant?

Reply:
`stat_prefix` is the conventional name in the config protos.

```
$ grep -R stat_prefix api/envoy/*
api/envoy/api/v2/core/grpc_service.proto:    string stat_prefix = 4 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/listener/v3/listener.proto:  // `listener.<stat_prefix>.`.
api/envoy/config/listener/v3/listener.proto:  string stat_prefix = 28;
api/envoy/config/core/v3/grpc_service.proto:    string stat_prefix = 4 [(validate.rules).string = {min_len: 1}];
api/envoy/config/filter/udp/udp_proxy/v2alpha/udp_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/rbac/v2/rbac.proto:  string stat_prefix = 3 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/tcp_proxy/v2/tcp_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/mongo_proxy/v2/mongo_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/zookeeper_proxy/v1alpha1/zookeeper_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/redis_proxy/v2/redis_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto:  string stat_prefix = 2 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/dubbo_proxy/v2alpha1/dubbo_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/ext_authz/v2/ext_authz.proto:  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/local_rate_limit/v2alpha/local_rate_limit.proto:  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/kafka_broker/v2alpha1/kafka_broker.proto:  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/rate_limit/v2/rate_limit.proto:  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/client_ssl_auth/v2/client_ssl_auth.proto:  string stat_prefix = 2 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/mysql_proxy/v1alpha1/mysql_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];
api/envoy/config/filter/network/thrift_proxy/v2alpha1/thrift_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_bytes: 1}];
api/envoy/extensions/matching/input_matchers/ip/v3/ip.proto:  string stat_prefix = 2 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/udp/dns_filter/v3/dns_filter.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/udp/udp_proxy/v3/udp_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/rbac/v3/rbac.proto:  string shadow_rules_stat_prefix = 5;
api/envoy/extensions/filters/network/rbac/v3/rbac.proto:  string stat_prefix = 3 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/local_ratelimit/v3/local_rate_limit.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/connection_limit/v3/connection_limit.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/tcp_proxy/v3/tcp_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/mongo_proxy/v3/mongo_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/zookeeper_proxy/v3/zookeeper_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto:  string stat_prefix = 2 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/dubbo_proxy/v3/dubbo_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/ext_authz/v3/ext_authz.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/ratelimit/v3/rate_limit.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/client_ssl_auth/v3/client_ssl_auth.proto:  string stat_prefix = 2 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/meta_protocol_proxy/v3/meta_protocol_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/network/thrift_proxy/v3/thrift_proxy.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/http/rbac/v3/rbac.proto:  string shadow_rules_stat_prefix = 3;
api/envoy/extensions/filters/http/local_ratelimit/v3/local_rate_limit.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/http/compressor/v3/compressor.proto:  //    `<stat_prefix>.compressor.<compressor_library.name>.<compressor_library_stat_prefix>.response.*`
api/envoy/extensions/filters/http/compressor/v3/compressor.proto:  //    `<stat_prefix>.compressor.<compressor_library.name>.<compressor_library_stat_prefix>.*`.
api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto:  //         stat_prefix: waf # This emits ext_authz.waf.ok, ext_authz.waf.denied, etc.
api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto:  //         stat_prefix: blocker # This emits ext_authz.blocker.ok, ext_authz.blocker.denied, etc.
api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto:  string stat_prefix = 13;
api/envoy/extensions/filters/http/bandwidth_limit/v3/bandwidth_limit.proto:  string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto:  string stat_prefix = 8;
```

Reply:
Maybe add a comment here that this is based on precedent but isn't guaranteed to be exhaustive? We might need to expand this in the future?